src/machine/mod.rs

```diff
@@ -70,6 +70,7 @@ lazy_static! {
 }
 
 #[derive(Debug)]
+#[repr(C)]
```

Comment:
You don't actually need `Machine` to be `repr(C)`, because it is always behind a pointer.